### ISSUE TITLE
SDL 2.28 fails to build with libdecor 0.1.1

### ISSUE THREAD

**Issue by tranzystorek-io** [libsdl-org/SDL#7891]
When building SDL 2.28 on Void Linux it fails with:

```
[285/587] Building C object CMakeFiles/SDL2.dir/src/video/wayland/SDL_waylanddyn.c.o
FAILED: CMakeFiles/SDL2.dir/src/video/wayland/SDL_waylanddyn.c.o
/usr/bin/ccache /usr/bin/cc -DSDL2_EXPORTS -DUSING_GENERATED_CONFIG_H -I/builddir/SDL2-2.28.0/build/include -I/builddir/SDL2-2.28.0/build/include/SDL2 -I/builddir/SDL2-2.28.0/build/include-config-none/SDL2 -I/builddir/SDL2-2.28.0/build/include-config-none -I/usr/include/libdrm -I/builddir/SDL2-2.28.0 -I/builddir/SDL2-2.28.0/build/wayland-generated-protocols -I/usr/include/libdecor-0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -DNDEBUG -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2    -ffile-prefix-map=/builddir/SDL2-2.28.0/build=. -idirafter "/builddir/SDL2-2.28.0/src/video/khronos" -DHAVE_LINUX_VERSION_H  -Wall -fno-strict-aliasing -Wdeclaration-after-statement -fvisibility=hidden -Wshadow -Wno-unused-local-typedefs -fdiagnostics-color=always -mmmx -m3dnow -msse -msse2 -msse3 -D_REENTRANT -I/usr/include/pipewire-0.3 -I/usr/include/spa-0.2 -I/usr/include -I/usr/include/libdrm  -DSDL_BUILD_MAJOR_VERSION=2 -DSDL_BUILD_MINOR_VERSION=28 -DSDL_BUILD_MICRO_VERSION=0 -fPIC -MD -MT CMakeFiles/SDL2.dir/src/video/wayland/SDL_waylanddyn.c.o -MF CMakeFiles/SDL2.dir/src/video/wayland/SDL_waylanddyn.c.o.d -o CMakeFiles/SDL2.dir/src/video/wayland/SDL_waylanddyn.c.o -c /builddir/SDL2-2.28.0/src/video/wayland/SDL_waylanddyn.c
../src/video/wayland/SDL_waylandsym.h: In function 'SDL_WAYLAND_LoadSymbols':
../src/video/wayland/SDL_waylandsym.h:221:27: error: 'libdecor_frame_get_min_content_size' undeclared (first use in this function); did you mean 'libdecor_frame_set_min_content_size'?
  221 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_min_content_size, (struct libdecor_frame *,\
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/video/wayland/SDL_waylanddyn.c:169:60: note: in definition of macro 'SDL_WAYLAND_SYM_OPT'
  169 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = fn;
      |                                                            ^~
../src/video/wayland/SDL_waylandsym.h:221:27: note: each undeclared identifier is reported only once for each function it appears in
  221 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_min_content_size, (struct libdecor_frame *,\
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/video/wayland/SDL_waylanddyn.c:169:60: note: in definition of macro 'SDL_WAYLAND_SYM_OPT'
  169 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = fn;
      |                                                            ^~
../src/video/wayland/SDL_waylandsym.h:224:27: error: 'libdecor_frame_get_max_content_size' undeclared (first use in this function); did you mean 'libdecor_frame_set_max_content_size'?
  224 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_max_content_size, (struct libdecor_frame *,\
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/video/wayland/SDL_waylanddyn.c:169:60: note: in definition of macro 'SDL_WAYLAND_SYM_OPT'
  169 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = fn;
      |                                                            ^~
```

these functions are missing from the libdecor API in 0.1.1

**Comment by Kontrabant** [comment]
This was fixed in SDL3 in https://github.com/libsdl-org/SDL/commit/5b5b67df203e9f8391e7e53aa5de54abf14c482e, but was never backported to SDL2.

Need to cherry-pick this commit.

**Comment by Kontrabant** [comment]
Fixed in the SDL2 and release-2.28.x branches.
